### PR TITLE
Dynamo streaming support

### PR DESCRIPTION
**Pull request by jagoda**
Provides the ability to perform streaming queries and streaming batch writes.

### CODE REVIEW

File: test/dynamodb/stream-query.test.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+  const items = [
+    { id: 'alpha', user: 'alice' },
+    { id: 'bravo', user: 'bob' },
+    { id: 'charlie', user: 'alice' }
+  ];
+
+  const querySpy = sinon.spy(documentClient, 'query');
+
+  await documentClient.batchWrite({
```
Comment (mdlavin) [inline test/dynamodb/stream-query.test.js]: It's an edge case, but you might want to check `UnprocessedItems` is empty to avoid surprises

File: test/dynamodb/stream-write.test.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    .map((item, index) => ({
+      item: { id: uuid() },
+      table: index % 2 ? oddTable : evenTable
+    }));
+
+  const {write, writeBufferSpace} = dynamodb.write(dynamoClient);
+
+  const expected = [];
+  const performed = await mockItemQueryStream(items, writeBufferSpace)
+    .pipe(operator)
```
Comment (mdlavin) [inline test/dynamodb/stream-write.test.js]: Any reason you used so many calls to `.pipe` in a row instead of the multi-arg pipe that other code uses?

File: test/dynamodb/stream-query.test.js (hunk 1 above)
Comment (jagoda) [reply test/dynamodb/stream-query.test.js]: Good call. Thanks.

File: test/dynamodb/stream-write.test.js (hunk 2 above)
Comment (jagoda) [reply test/dynamodb/stream-write.test.js]: Nope. It was just already this way and was working. Should be easy enough to cleanup.
